[MIG] purchase_order_type_ux: Migration to 19.0

migration-bot-adhoc opened the pull request:
Simplificamos mucho la funcionalidad:
1. el tipo de ventas en facturas de compra agrega mucha complejidad y no agrega mucho valor
2. implementamos la compatibilidad con branches como hicimos para sale_order_type
3. todo lo que tiene que ver con reporte argentino (report_partner_id) lo sacamos porque debería ir en otro módulo pero además creemos que con nuevo approach a/b no debería ser muy necesario
4. feature de setear partner y project no tiene tanto sentido (project no lo tenía nadie y partner debería setearse el tipo en el partner, solo lo usaba un cliente)

migration-bot-adhoc committed:
[MIG] purchase_order_type_ux: Migration to 19.0

roboadhoc commented:
[![Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/purchase/pull/286.png)](https://mergebot.adhoc.ar/ingadhoc/purchase/pull/286)

jjscarafia committed:
[IMP] _order_type_ux: simplify module

Simplificamos mucho la funcionalidad:
1. el tipo de ventas en facturas de compra agrega mucha complejidad y no agrega mucho valor
2. implementamos la compatibilidad con branches como hicimos para sale_order_type
3. todo lo que tiene que ver con reporte argentino (report_partner_id) lo sacamos porque debería ir en otro módulo pero además creemos que con nuevo approach a/b no debería ser muy necesario
4. feature de setear partner y project no tiene tanto sentido (project no lo tenía nadie y partner debería setearse el tipo en el partner, solo lo usaba un cliente)
